docs/source/developers/python.rst

```diff
@@ -55,12 +55,15 @@ like so:
 
 .. code-block:: shell
 
-   pytest pyarrow
+   pytest arrow/python/pyarrow
 
 Package requirements to run the unit tests are found in
 ``requirements-test.txt`` and can be installed if needed with ``pip install -r
 requirements-test.txt``.
 
+If the test fails to connect to libarrow run ``python -m pytest pyarrow`` and
+check if the editable version of pyarrow was installed correctly.
```

Comment:
"connect to" isn't really a faithful representation of what happens here. Perhaps:

"""If you get import errors for `pyarrow._lib` or another PyArrow module when trying to run the tests, check if..."""